workloads/uber/src/main/java/io/vectorized/tx_compact/Workload.java

```diff
@@ -0,0 +1,590 @@
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import java.lang.Thread;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Collections;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import java.time.Duration;
+import java.util.Random;
+import java.util.concurrent.Future;
+
+import java.nio.charset.Charset;
+
+public class Workload {
+    public volatile boolean is_active = false;
+
+    private volatile App.InitBody args;
+    private BufferedWriter opslog;
+    private volatile Random random = new Random();
+
+    private HashMap<Integer, App.OpsInfo> ops_info;
+    private synchronized void succeeded(int thread_id) {
+        ops_info.get(thread_id).succeeded_ops += 1;
+    }
+    private synchronized void failed(int thread_id) {
+        ops_info.get(thread_id).failed_ops += 1;
+    }
+
+    private volatile long last_offset = -1;
+
+    private long last_error_id = 0;
+    private synchronized long get_error_id() {
+        return ++this.last_error_id;
+    }
+
+    private long past_us;
+    private synchronized void log(int thread_id, String message) throws Exception {
+        var now_us = System.nanoTime() / 1000;
+        if (now_us < past_us) {
+            throw new Exception("Time cant go back, observed: " + now_us + " after: " + past_us);
+        }
+        opslog.write("" + thread_id +
+                        "\t" + (now_us - past_us) +
+                        "\t" + message + "\n");
+        past_us = now_us;
+    }
+    private synchronized void violation(int thread_id, String message) throws Exception {
+        var now_us = System.nanoTime() / 1000;
+        if (now_us < past_us) {
+            throw new Exception("Time cant go back, observed: " + now_us + " after: " + past_us);
+        }
+        opslog.write("" + thread_id +
+                        "\t" + (now_us - past_us) +
+                        "\tviolation" +
+                        "\t" + message + "\n");
+        opslog.flush();
+        opslog.close();
+        System.exit(1);
+        past_us = now_us;
+    }
+    public void event(String name) throws Exception {
+        log(-1, "event\t" + name);
+    }
+
+    private volatile ArrayList<Thread> producers;
+    private volatile ArrayList<Thread> edge_consumers;
+    private volatile ArrayList<Thread> hist_consumers;
+    
+    private HashMap<Integer, Integer> last_committed_seq;
+    private HashMap<Integer, Integer> committing_ltid;
+
+    public Workload(App.InitBody args) {
+        this.args = args;
+    }
+
+    private String randomString(Random random, int length) {
```

Comment:
This seems to generate a lot of non-ascii chars that are difficult to parse/analyze while debugging. Rather I did the following diff that made my life easier, consider applying this patch.

```
diff --git a/workloads/uber/src/main/java/io/vectorized/tx_compact/Workload.java b/workloads/uber/src/main/java/io/vectorized/tx_compact/Workload.java
index 05f33ee..a44ed6b 100644
--- a/workloads/uber/src/main/java/io/vectorized/tx_compact/Workload.java
+++ b/workloads/uber/src/main/java/io/vectorized/tx_compact/Workload.java
@@ -82,6 +82,8 @@ public class Workload {
     
     private HashMap<Integer, Integer> last_committed_seq;
     private HashMap<Integer, Integer> committing_ltid;
+    private static final String randAlpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    private static final int alphaLen = randAlpha.length();
 
     public Workload(App.InitBody args) {
         this.args = args;
@@ -89,12 +91,12 @@ public class Workload {
 
     private String randomString(Random random, int length) {
         byte[] array = new byte[length];
-        random.nextBytes(array);
-        for (int i=0;i<length;i++) {
-            // 32 is ' ', 126 is '~', generating all in between
-            array[i] = (byte)(32 + array[i]%(126 - 32));
+       StringBuffer buf = new StringBuffer();
+       buf.append('~');
+        for (int i=0;i<length-1;i++) {
+         buf.append(randAlpha.charAt(random.nextInt(alphaLen)));
         }
-        return new String(array, Charset.forName("UTF-8"));
+        return buf.toString();
     }
 
     public void start() throws Exception {
@@ -587,4 +589,4 @@ public class Workload {
             }
         }
     }
-}
\ No newline at end of file
+}
```